src/extension.ts

```diff
@@ -181,20 +181,30 @@ class ExtensionManager implements vscode.Disposable {
     telemetry.logEvent('open', telemetryProperties);
   }
 
+  public getActiveFolderContext() : StateManager | undefined {
+    const activeFolder = this.getActiveFolder();
+    if (activeFolder) {
+      const extensionState = new StateManager(this.extensionContext, activeFolder);
+      return extensionState;
+    }
+
+    return undefined;
+  }
+
   // Partial activation means that the CMake Tools commands are hidden
   // from the commands pallette and the status bar is not visible.
   // The context variable "cmake:enableFullFeatureSet" is always equal
   // to the state setting ignoreCMakeListsMissing.
   // To have them both always in sync, cmake:enableFullFeatureSet is set
   // by the getter and setter of ignoreCMakeListsMissing.
-  public async enableFullFeatureSet(fullFeatureSet: boolean) {
-    const activeFolder = this.getActiveFolder();
-    if (activeFolder) {
-      const extensionState = new StateManager(this.extensionContext, activeFolder);
-      extensionState.ignoreCMakeListsMissing = !fullFeatureSet;
+  public enableFullFeatureSet(fullFeatureSet: boolean) {
+    const context = this.getActiveFolderContext();
+    if (context) {
+      context.ignoreCMakeListsMissing = !fullFeatureSet;
+      this._statusBar.setVisible(fullFeatureSet);
+    } else {
+      log.info(localize('', 'enableFullFeatureSet(\'{fullFeatureSet}\') called but not active folder found.'));
```

Comment:
Did you mean to insert the value of `fullFeatureSet` here?  You need to use `{0}` and pass the parameter in afterwards:
```
localize('enableFullFeatureSet.no.active.folder', 'enableFullFeatureSet({0}) called but no active folder found.', fullFeatureSet);
```

Note: I also fixed a spelling mistake in the message.

Reply:
And actually, I don't think this message belongs in the `info` stream.  I think `log.error` is more appropriate.

Reply:
Oh, I made a confusion with some other scenario. And I forgot to test that message.